### PR TITLE
kubelet: sysctls allows slashes as a separator

### PR DESCRIPTION
**Pull request by mengjiao-liu**
Sysctl allows slashes as a separator in kubelet.

ref https://github.com/kubernetes/kubernetes/pull/102393
ref https://man7.org/linux/man-pages/man5/sysctl.d.5.html

### CODE REVIEW

**Comment by mengjiao-liu** [comment]
Please take a look  @mrunalp @derekwaynecarr

File: content/en/docs/tasks/administer-cluster/sysctl-cluster.md [hunk 1]
```diff
@@ -53,6 +53,14 @@ same node. This means that setting a _safe_ sysctl for one pod
 - must not allow to gain CPU or memory resources outside of the resource limits
   of a pod.
 
+{{< note >}}
+Starting from Kubernetes version 1.23, kubelet supports the use of "/" or "." as separators for sysctl.
+Sysctls can be converted to a slash as a separator.
+For example, the sysctl `kernel.shm_rmid_forced` can also be written as `kernel/shm_rmid_forced` with a slash as a separator.
+For more sysctl parameter conversion method details, please refer to
+[Linux sysctl documentation](https://man7.org/linux/man-pages/man5/sysctl.d.5.html)
```
Comment (sftim) [inline content/en/docs/tasks/administer-cluster/sysctl-cluster.md:61]:
```suggestion
the page [sysctl.d(5)](https://man7.org/linux/man-pages/man5/sysctl.d.5.html) from
the Linux man-pages project.
```

I'd consider linking to https://www.freedesktop.org/software/systemd/man/sysctl.d.html in the systemd docs instead. (I'm not sure which Linux package owns this convention).

File: content/en/docs/tasks/administer-cluster/sysctl-cluster.md [hunk 2]
```diff
@@ -53,6 +53,14 @@ same node. This means that setting a _safe_ sysctl for one pod
 - must not allow to gain CPU or memory resources outside of the resource limits
   of a pod.
 
+{{< note >}}
+Starting from Kubernetes version 1.23, kubelet supports the use of "/" or "." as separators for sysctl.
+Sysctls can be converted to a slash as a separator.
+For example, the sysctl `kernel.shm_rmid_forced` can also be written as `kernel/shm_rmid_forced` with a slash as a separator.
```
Comment (sftim) [inline content/en/docs/tasks/administer-cluster/sysctl-cluster.md:59]:
```suggestion
For example, you can represent the same sysctl name as `kernel.shm_rmid_forced` using a
period as the separator, or as `kernel/shm_rmid_forced` using a slash as a separator.
```

File: content/en/docs/tasks/administer-cluster/sysctl-cluster.md [hunk 3]
```diff
@@ -53,6 +53,14 @@ same node. This means that setting a _safe_ sysctl for one pod
 - must not allow to gain CPU or memory resources outside of the resource limits
   of a pod.
 
+{{< note >}}
+Starting from Kubernetes version 1.23, kubelet supports the use of "/" or "." as separators for sysctl.
```
Comment (sftim) [inline content/en/docs/tasks/administer-cluster/sysctl-cluster.md:57]:
```suggestion
Starting from Kubernetes version 1.23, the kubelet supports the use of either `/` or `.`
as separators for sysctl names.
```

**Review by sftim — CHANGES_REQUESTED** [review]
I'd move this new note into the page overview (line 15). It's not specific to unsafe sysctls.

**Comment by mengjiao-liu** [comment]
> I'd move this new note into the page overview (line 15). It's not specific to unsafe sysctls.

Because now in k8s, `Setting Sysctls for a Pod` and `PodSecurityPolicy` part does not support sysctl to contain slashes.
Only kubelet supports this first, and the API verification support sysctl including the slash is expected to be completed in version 1.24.  I am worried that it may cause ambiguity, so I did not put this attention part in overview.

Maybe, I put this part in the overview, but in the following `Setting Sysctls for a Pod` and `PodSecurityPolicy` part, write some explanation? @sftim

File: content/en/docs/tasks/administer-cluster/sysctl-cluster.md (hunk 1 above)
Comment (mengjiao-liu) [reply content/en/docs/tasks/administer-cluster/sysctl-cluster.md:61]: This looks more accurate, thanks.

File: content/en/docs/tasks/administer-cluster/sysctl-cluster.md (hunk 2 above)
Comment (mengjiao-liu) [reply content/en/docs/tasks/administer-cluster/sysctl-cluster.md:59]: Thanks for your suggestion, I will update it.

File: content/en/docs/tasks/administer-cluster/sysctl-cluster.md (hunk 3 above)
Comment (mengjiao-liu) [reply content/en/docs/tasks/administer-cluster/sysctl-cluster.md:57]: Thanks for your suggestion, I will update it.

**Review by sftim — APPROVED** [review]
/approve

**Comment by sftim** [comment]
/sig node

We should get tech review from SIG Node to confirm that the v1.23 code matches what the documentation will say (should be mostly a formality).

**Comment by mengjiao-liu** [comment]
> We should get tech review from SIG Node to confirm that the v1.23 code matches what the documentation will say (should be mostly a formality).

Request for review. @sig-node-reviewers

File: content/en/docs/tasks/administer-cluster/sysctl-cluster.md [hunk 4]
```diff
@@ -13,6 +13,17 @@ This document describes how to configure and use kernel parameters within a
...
+{{< note >}}
+Starting from Kubernetes version 1.23, the kubelet supports the use of either `/` or `.`
+as separators for sysctl names. 
+For example, you can represent the same sysctl name as `kernel.shm_rmid_forced` using a
+period as the separator, or as `kernel/shm_rmid_forced` using a slash as a separator.
+For more sysctl parameter conversion method details, please refer to
+the page [sysctl.d(5)](https://man7.org/linux/man-pages/man5/sysctl.d.5.html) from
+the Linux man-pages project.
+Setting Sysctls for a Pod and PodSecurityPolicy features do not yet support 
+setting sysctls with slashes.
```
Comment (pacoxu) [inline content/en/docs/tasks/administer-cluster/sysctl-cluster.md:25]: Should we mention that `Setting Sysctls for a Pod and PodSecurityPolicy features will support slashes in 1.24 and later`?

The reason should be APIServer need to be compatible with 1.n-1 version kubelet.

Comment (pacoxu) [reply content/en/docs/tasks/administer-cluster/sysctl-cluster.md:25]: The description is correct to me. LGTM.

Comment (mengjiao-liu) [reply content/en/docs/tasks/administer-cluster/sysctl-cluster.md:25]: I will update website PR after 1.24 adds this feature.

Comment (pacoxu) [reply content/en/docs/tasks/administer-cluster/sysctl-cluster.md:25]: /lgtm

**Pull request merged**